include/fx/AnimateableTimeStretchPitchScaleReader.h

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright 2009-2025 Jörg Müller
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+#pragma once
+
+/**
+ * @file AnimateableTimeStretchPitchScaleReader.h
+ * @ingroup fx
+ * The AnimateableTimeStretchPitchScaleReader class.
+ */
+
+#include "fx/AnimateableTimeStretchPitchScale.h"
+#include "fx/EffectReader.h"
+#include "fx/TimeStretchPitchScaleReader.h"
+
+using namespace RubberBand;
+
+AUD_NAMESPACE_BEGIN
+
+/**
+ * This class reads from another reader and applies time-stretching and pitch scaling with support for animating both properties.
+ */
+class AUD_API AnimateableTimeStretchPitchScaleReader : public EffectReader
+{
+private:
+	/**
+	 * The animateable time stretch, pitch scale effect
+	 */
+	AnimateableTimeStretchPitchScale* m_timeStretchPitchScale;
```

Comment:
It should be possible to use the `AnimateableTimeStretchPitchScaleReader` without having to keep the `AnimateableTimeStretchPitchScale` alive, or even without ever allocating it.

This means that all the parameters you have in the `AnimateableTimeStretchPitchScale` need to be passed to this class. For the `AnimateableProperty` members, in order to share them and be able to update them, you shoud store them in an `std::shared_ptr`.